tests/functional/dbt_contracts/dbt_contracts.yml

```diff
@@ -0,0 +1,90 @@
+# dbt_contracts.yml
+# upstream-only config for illustration
+consumer:
+  core-only: # give the project a plain name to ref
+    path: git repo OR local subfolder # example: "https://{{env_var('DBT_ENV_SECRET_GIT_CREDENTIAL')}}@github.com/dbt-labs/awesome_repo.git" OR ./models/core/
+    version: 0.2.0 # Versioning is at the project level, NOT the model level
+    models:
+      - ref('fct_orders')
+      - ref('dim_customers')
```

Comment:
@callum-mcdata make a code suggestion for metrics/entities!

Reply:
I think this is tbd depending on how we implement metrics/entities in the future. But my thinking after this weekend is that `entity` would be a type of model that is contains more information than a base model type. So I believe it could look like this:

```
    models:
      - entity('orders')
      - ref('dim_customers')
    metrics:
      - metric('revenue')
 
```

In the above example, we would have an entity defined on top of orders that is what we want to expose in downstream dags or consuming tools. The metric of revenue would then be dependent on orders.

Reply:
This is a great read on code contracts as a whole: https://gist.github.com/chitchcock/1281611